### ISSUE TITLE
RCS1032 False positive targeting "F(G<A,B>(7));" grammar ambiguity

### ISSUE THREAD

**Issue by sgkoishi** [dotnet/roslynator#659]
**Product and Version Used**: 
Roslynator 2.9.0, Visual Studio 2019 16.4.4

**Steps to Reproduce**:
[Demo on SharpLab](https://sharplab.io/#v2:CYLg1APgAgTAjAWAFBQMwAJboMLoN7LpGYZRwBsmALOgGIAUARgPbMA26jANJ6x4zACUhYgSTF0AXxFE01dAFl6g9DPxqJASwB2AF3TaArgFt0AXnQAGANwbiO/UeMxzV2+InEAIumCujbGzunsQA9KEMTugAPAYmMDx+AHzo9H4A/Om+goLBIXT09FGxTkKJ6Clp6JnZuWrSSA1yWF7IYhLNFLzs6MwADgCmAE4AhrrMQxX0PmwDAGa6PD5DmgDmABa6wh5E7fm660PMAO4GA6cAcsy6AJLGfbPGA3oDwACiAB4AxgN9uprMbTKPJEBodUhdFg9frDMYTGLTdCzBZLdArDZbNR7EIHI6nbTndBXW73R7PXSvT4/P4AoF1HZSZCSIA==)
Details: icsharpcode/ILSpy#1959 ([`F(G<A,B>(7));` grammar ambiguity on MSDN](https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/proposals/csharp-7.0/pattern-matching#changes-to-syntactic-disambiguation))

**Actual Behavior**:
`RCS1032	Remove redundant parentheses.` shown for `(num < num2)`

**Expected Behavior**:
The parentheses are necessary to avoid ambiguity.

**Comment by JosefPihrt** [comment]
This is a rare case that will be hard (or impossible) to detect during code analysis. I think it is reasonable to suppress this diagnostic one by one.

Example:
```cs
#pragma warning disable RCS1032
#pragma warning restore RCS1032
```

**Comment by dgrunwald** [comment]
[My solution for ILSpy](https://github.com/icsharpcode/ILSpy/blob/0cf50aa8277c911cf53d2864c98f6751bfb0b4db/ICSharpCode.Decompiler/CSharp/OutputVisitor/GenericGrammarAmbiguityVisitor.cs) actually works on the syntax tree; essentially trying to re-interpret the existing expression syntax nodes as if they were types.
A similar approach should be possible here.
